### PR TITLE
Changelog for 2.0 release.

### PR DESCRIPTION
**Pull request by phil-levis**
### Pull Request Overview

This pull request updates the CHANGELOG for 2.0.

### Testing Strategy

N/A

### TODO or Help Wanted

This pull request still needs additional information that you think is worth being in the Changelog.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,128 @@
...
+	
+	* ESP32-C3-DevKitM-1
+	
+
+* Major HIL changes
+  
+  * All HILs have changed significantly, to be in line with the new
+  types within the kernel.
+  
+  * `ReturnCode` has been deprecated within the kernel. HILs that used to
```
Comment (hudson-ayers) [inline CHANGELOG.md:88]:
```suggestion
  * `ReturnCode` has been removed from the kernel. HILs that used to
```

**Review by hudson-ayers — COMMENTED** [review]
Additional items:

- Changes to kernel internals and the build system to reduce code size
- Update all non-virtualized userspace drivers to use grant regions and to forbid access by multiple apps
- Add utilities for analyzing code size and stack usage
- update inline assembly to the new, better checked `asm!()` macro from the deprecated `llvm_asm!()`
- Make it possible for boards to avoid paying the code size cost for peripherals they do not use (https://github.com/tock/tock/pull/2069)
- Bug fixes

**Comment by phil-levis** [comment]
> Changes to kernel internals and the build system to reduce code size

Can you point me at some specific PRs?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,128 @@
+New in 2.0
+==========
+
+* Many core kernel APIs have been redesigned and rewritten.
+  
+  * There are two new userspace system calls: `AllowReadOnly` and
+  `Exit`. The old `Allow` system call has been renamed to `AllowReadWrite`.
+  `AllowReadyOnly` provides a mechanism for userspace to share a
```
Comment (jrvanwhy) [inline CHANGELOG.md:8]: `AllowReadyOnly` -> `AllowReadOnly`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,128 @@
...
+  the format and structure of the allowed return types. TRD104 documents
+  the new ABI.
+  
+  * The calling semantics and requirements for system calls is more
+  clearly defined, especially with respect to calls to the Allow
+  system calls and how buffers are managed. Furthermore, the lifetime
+  of upcalls passed to the kernel with the `subscribe` system call
+  has been defined. To enforce that the kernel doesn't maintain
+  references to upcalls that it shouldn't (so userspace can reclaim any
+  resources they require), upcalls are now managed by the kernel.
```
Comment (jrvanwhy) [inline CHANGELOG.md:24]: "upcalls are now managed by the kernel" -> "upcalls are now managed by the core kernel"

**Comment by hudson-ayers** [comment]
> > Changes to kernel internals and the build system to reduce code size
> 
> Can you point me at some specific PRs?

https://github.com/tock/tock/pull/2648

https://github.com/tock/tock/pull/2639

https://github.com/tock/tock/pull/2606

https://github.com/tock/tock/pull/2472

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,153 @@
...
+  * Almost everything is now exported as `kernel::module::Type` rather than `kernel::Type`.
+  * `/common` is split up into `/utilities` and `/collections`
+	
+* There is increased chip and board support.
+
+  * RISC-V support has been extended to support progress and revisions
+  to support microcontroller-type systems, including support for EPMP
+  memory protection.
+  
+  * There is support for ARM CortexM7.
```
Comment (alexandruradovici) [inline CHANGELOG.md:59]:
```suggestion
  * There is support for ARM CortexM0+ and CortexM7.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,153 @@
...
+  Now, if a structure implements both `uart::Transmit` and 
+  `uart::Receive`, it will automatically implement `uart::UartData`.
+  
+* In-kernel virtualizers for the following HILs have been added: AES, RNG, SHA
+  
+* The kernel now checks whether loaded processes are compiled for the
+running kernel version. Because 2.0 changes the user/kernel ABI, processes
+compiled for Tock 1.x will not run correctly on a Tock 2.x kernel and
+vice versa. If the kernel detects that a process is compiled for the
+wrong kernel version it skips the process and does not load it.
```
Comment (alexandruradovici) [inline CHANGELOG.md:127]:
```suggestion
wrong kernel version it stops loading processes.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,153 @@
...
+  * Digest has been extended to support multiple digest algorithms:
+  in addition to HMAC SHA256 it now supports SHA224, SHA256, SHA384, SHA512,
+  HMAC SHA384 and HMAC SHA512.
+
+  * A key/value store HIL has been added.
+  
+  * An 8080 bus HIL (for LCDs) has been added.
+  
+  * A text screen HIL has been added.
+
```
Comment (alexandruradovici) [inline CHANGELOG.md:112]:
```suggestion
  * A screen HIL has been added
  
  * A touch HIL has been added
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,153 @@
...
+wrong kernel version it skips the process and does not load it.
+
+* There have been changes to kernel internals and the build system
+to reduce code size. For example, kernel code that was highly 
+replicated in monomorphized functions has been factored out (#2648).
+
+* All system call driver capsules that do not support use by 
+multiple processes now use grant regions to store state and
+explicitly forbid access from multiple processes (e.g., #2518).. 
+
```
Comment (alexandruradovici) [inline CHANGELOG.md:136]:
```suggestion

  * The process console has been improved

```

**Review by bradjc — DISMISSED** [review]

**Review by bradjc — APPROVED** [review]

**Pull request merged**